basics/trey-hunner-zip-and-enumerate.md

```diff
@@ -75,71 +75,49 @@ b 2
-iterators will be joined together once the function is executed.
-
-### Example 1
-
-Let's look at an example, suppose we wish to join 2 tuples together.
-
 ```python
->>> users = ("Tushar", "Aman", "Anurag", "Sohit")
-uids = ("usr122", "usr123", "usr124", "usr125")
-user_details = zip(uids, users)
-print(user_details)
->>>
-<zip object at 0x149648b0d5c0>
+>>> users = ["Tushar", "Aman", "Anurag", "Sohit"]
+>>> uids = ["usr122", "usr123", "usr124", "usr125"]
+>>> user_details = zip(uids, users)
+>>> print(user_details)
+<zip object at 0x0000016938E311C0>
 >>>
 ```
 
-If we want to get some real output we can convert this zip object into List, tuple or a set.
-
-### Example 2
+After calling zip, an iterator is returned. In order to see the content wrapped inside, we need to first convert it to a list. Basically, `zip()` is an iterator, i.e. lazy: it gives the items as needed, instead of calculating them and storing them into memory all at once. So the zip object cannot show its elements before the elements are used, because it hasn't computed them yet.
 
 ```python
->>> users = ("Tushar", "Aman", "Anurag", "Sohit")
-uids = ("usr122", "usr123", "usr124", "usr125")
-user_details = zip(uids, users)
-print(list(user_details)) # converting to list
->>>
+>>> users = ["Tushar", "Aman", "Anurag", "Sohit"]
+>>> uids = ["usr122", "usr123", "usr124", "usr125"]
+>>> user_details = zip(uids, users)
+>>> print(list(user_details))
 [('usr122', 'Tushar'), ('usr123', 'Aman'), ('usr124', 'Anurag'), ('usr125', 'Sohit')]
 >>>
 ```
 
-What if these iterators were of different lengths? well, then the ones with the extra fields will be ignored like so.
-
-### Example 3
+If the lists are of different lengths, some items from the end of the longer list will be ignored.
 
 ```python
->>> users = ("Tushar", "Aman", "Anurag", "Sohit") 
-uids = ("usr122", "usr123", "usr124") 
-emails = ("tushar@example.com", "aman@example.com", "anurag@example.com", "sohit@example.com")
-user_details = zip(uids, users, emails)
-print(list(user_details))
->>>
-[('usr122', 'Tushar', 'tushar@example.com'), ('usr123', 'Aman', 'aman@example.com'), ('usr124', 'Anurag', 'anurag@example.com')]
+>>> users = ["Tushar", "Aman", "Anurag", "Sohit"]
+>>> emails = ["tushar@example.com", "aman@example.com", "anurag@example.com", "sohit@example.com"]
+>>> users_contact = zip(users, emails)
+>>> print(list(users_contact))
+[('Tushar', 'tushar@example.com'), ('Aman', 'aman@example.com'), ('Anurag', 'anurag@example.com')]
 >>>
 ```
 
-Length of `uids` is 3 thus it is the shortest tuple, now `zip()` function will crop other tuples to consider them equally and will finally join all of them together.
+Length of `users` list is 3 thus it is the shortest. `zip()` function will crop other lists to consider them equally and will finally join all items together.
 
-### Example 4: Using zip with for loop
+### Using zip with for loop
```

Comment:
Before moving onto a new section, we should probably mention that it's usually a mistake to call `zip()` with lists of different lengths. I don't think I have ever intentionally zipped lists of different lengths.

Reply:
I think calling it a mistake is not viable, we should say not recommended instead!